setup/dns.sh

```diff
@@ -45,7 +45,11 @@ EOF
 # DNS queries that require a recursive nameserver, and the system
 # might have other network interfaces for e.g. tunnelling, we have
 # to be specific about the network interfaces that nsd binds to.
-for ip in $PRIVATE_IP $PRIVATE_IPV6; do
+#
+# NSD does not support providing the same ip multiple times so we 
+# have to sort through the list and make sure, that each ip is only
+# listed once in the config
+for ip in `echo $PUBLIC_IP $PRIVATE_IP $PUBLIC_IPV6 $PRIVATE_IPV6 | tr ' ' '\n' | sort -u`; do
```

Comment:
From https://www.nlnetlabs.nl/projects/nsd/nsd.conf.5.html:
ip-address: <ip4 or ip6>[@port]
              NSD will bind to the listed ip-address.  Can  be  give  multiple
              times  to  bind multiple ip-addresses. Optionally, a port number
              can be given.  If none are given NSD  listens  to  the  wildcard
              interface. Same as commandline option -a.  For servers with mul-
              tiple IP addresses that can be  used  to  send  traffic  to  the
              internet, list them one by one, or the source address of replies
              could be wrong.  This is because if the udp socket associates  a
              source  address  of  0.0.0.0 then the kernel picks an ip-address
              with which to send to the internet, and it picks the wrong  one.
              Typically  needed  for anycast instances.  Use ip-transparent to
              be able to list addresses that turn on later (typical  for  cer-
              tain load-balancing).

Since ip-transparent is also set in the mailinthebox config, giving additional ip addresses to nsd should not be an issue. In your case, this would make nsd listen on all the private interfaces (as it is currently doing anyway).